modules/learn/pages/security/using-multiple-cas.adoc

```diff
@@ -0,0 +1,72 @@
+
+. The certificate for Client 1 has been signed by an intermediate certificate, which has been directly signed by the root certificate _CA a_: this is the CA providing authority to Node 1 and Node 2.
+This allows mutual authentication to proceed between Node 1 and Client 1, since the certificate of each takes its authority from _CA a_.
+(Note that this is the only form of certificate-based authentication that was supported by Couchbase Server, prior to version 7.1.)
+
+. The certificate for Client 2 has been directly signed by the root certificate _CA h_.
+Client 2 also maintains the root certificate _CA a_ in its trust store.
+This allows mutual authentication to proceed between Node 2 and Client 2; since Client 2 recognizes the authority of _CA a_, and Node 1 recognizes the authority of _CA h_.
+In each case, the recognized root CA exists in the entity's trust store, but has not played a role in the signing of the entity's own certificate.
+
+. The certificate for Client 3 has been signed by an intermediate certificate, which has itself been signed by the root certificate _CA k_.
+Client 3 also maintains _CA e_ in its trust store.
+This prevents mutual authentication from occurring between Client 3 and Node 3: although the certificate for Node 3 acquires its authority from _CA e_, which is in the client's trust store, and is thus recognized by the client; the client's certificate is signed by _CA k_, which is not in the cluster's trust store, and is therefore not recognized by the cluster.
+
+. The certificate for Client 4 is directly signed by _CA c_.
+The client's trust store also contains _CA e_ and _CA g_.
+This allows mutual authentication to succeed between Client 4 and Node 3, since the authority of each is derived from a root certificate in its opposite's trust store.
+This also allows mutual authentication to succeed between Client 4 and Node 4, for the same reason.
+
+== Clusters Running Multiple Couchbase-Server Versions
+
+Multiple CAs are supported by Couchbase Server Version 7.1 and later.
+Typically, when a cluster is upgraded, it temporarily includes nodes of two versions; one being the version _to_ which the cluster is being upgraded, the other the version _from_ which it is being upgraded.
+
+Once the cluster's nodes are all upgraded to version 7.1 or later, multiple CAs are fully supported.
+During the upgrade period, only _some_ of the supportive APIs can be used: for information, see xref:rest-api:rest-certificate-management.adoc[Certificate Management API].
+
+== Adding Nodes
+
+When a new node is added to a Couchbase-Server cluster that is version 7.1 or later, the new node may be added with its existing root certificate and corresponding certificate chain, provided that the root certificate is already included in the cluster's trust store: if the node's certificate has not been added to the trust store, an error is flagged.
```

Comment:
We can also mention that new node's trust store is cleaned when the node is joining (not merged). To be more precise: new node's trusted store can be used for initial connection establishment (right before join) to the cluster (might be needed when join is initiated from the new node). But a bit later, when the node is actually joining the cluster, its trust store is overwritten with cluster's trust store. 

I'll try to explain in example:
Given: Node-to-be-added joins Cluster

Cluster's trust store: ca1, ca2, ca3
Cluster node1: cert1 signed by ca1

Node-to-be-added's trusted store: ca1, ca2, ca4
Node-to-be-added's node cert: cert2 signed by ca2

After join new cluster's trust store will contain: ca1, ca2, ca3 (stays the same basically).
ca4 will not be added.

Reply:
I have added.

A couple of questions. First, what if cert2 had been signed by CA 4?

Secondly (and this relates to a comment you have made elsewhere): there is a distinction between adding a node and joining a node. As a cluster, I appoint an external node that is to be added. As an external node, I appoint a cluster to which I join. But I believe that a join can only occur before provisioning - which would mean, before the addition of certificates. Therefore, a node with uploaded certs can only be added, not joined. Is that correct?

Reply:
> First, what if cert2 had been signed by CA 4?

Then addition will not work, because cluster doesn't trust CA4


> adding a node to a cluster' and 'joining a node to a cluster'. 

I think you mean that the "join" can be initiated either by node-to-be-added or by the cluster.
In other words you can start the process either from cluster, or from node-to-be-added. 
Technically both of them are almost the same. The only difference is one extra http call for the case when the "join" is initiated from node-to-be-added.


> But I believe that a join can only occur before provisioning - which would mean, before the addition of certificates.

Not sure what you call provisioning here.
There are two options:
1. If your cluster uses self-generated certs, then nothing needs to be done before joining.
2. If your cluster uses custom certs, you only need to load certificates to the node-to-be-added before joining (note that this is not equal to initializing the cluster or initializing node). Upload CA and node cert to the inbox dir and call /loadCAcerts and /reloadCertificate (CLI should be able to do that as well).

Those things are true with no regards if the addition is initiated from cluster side or from node-to-be-added side.

Reply:
Thank you. I have added a note on addition not working when cert 2 is signed by CA 4:

Note also that if the node certificate for _node x_, _Cert 2_, had been signed by _CA 4_ (instead of _CA 2_), the addition of _node x_ would fail; because the cluster does not trust _CA 4_.